### DISCUSSION TITLE
Running sqlite inmemory for a long time

### DISCUSSION THREAD

**Discussion by IgnisDa** [apalis-dev/apalis#164] [Q&A] [answered]
I am using the SQLite storage for my jobs (`sqlite::memory:`). I have noticed that when my server is running for a long time, the following error pops ups.
 
![image](https://github.com/geofmureithi/apalis/assets/60938164/c293b112-e975-41aa-aa64-372d421c3998)

The only way to fix it is to restart the server.

I am not sure where I should start debugging this since it does not happen in local development and only after 1-2 days in production.

**Comment by IgnisDa** [comment]
NOTE: It is imperative that the SQLite database is started in memory because of how the file system is set up in my project (which does not allow me to write the database to a file).

**Comment by geofmureithi** [comment]
This seems related to:
https://github.com/launchbadge/sqlx/issues/1647

**Reply by IgnisDa** [reply]
Thanks!

This does seem to have performance implications. Will have to monitor if it is significant.

**Comment by geofmureithi** [comment, marked as answer]
The solution
```rust
let inner = sqlx::pool::PoolOptions::new()
    .max_lifetime(None)
    .idle_timeout(None)
    .connect(&std::env::var("DATABASE_URL")?)
    .await?;
let storage = SqliteStorage::new(inner);
```
